DEVFILE_REGISTRY.md

```diff
@@ -61,3 +61,171 @@ If you want to send telemetry information to your own Segment instance, specify
+  name: my-devfile-registry
+spec:
+  devfileIndex:
+    image: quay.io/my-devfile/devfile-index:next
+    imagePullPolicy: Always
+  ociRegistry:
+    image: quay.io/my-devfile/oci-registry:next
+    imagePullPolicy: Always
+  registryViewer:
+    image: quay.io/my-devfile/registry-viewer:next
+    imagePullPolicy: Always
+EOF
+```
+
+## Disabling the web frontend 
+
+You can ask the operator to deploy the Devfile Registry without the `registry-viewer` container, by setting the field `spec.headless` to `true`.
+
+```bash
+$ cat <<EOF | oc apply -f -
+apiVersion: registry.devfile.io/v1alpha1
+kind: DevfileRegistry
+metadata:
+  name: headless-devfile-registry
+spec:
+  devfileIndex:
+    image: quay.io/devfile/devfile-index:next
+  telemetry:
+    registryName: test
+  headless: true
+EOF
+```
+
+## Using a Persistent Volume to store OCI images
+
+The Devfile stacks are stored in OCI images served by the OCI registry container. By default, an Ephemeral volume is created and used to store these images.
+
+You can ask the operator to create a Persistent Volume and to mount it into the `oci-registry` container's filesystem, so the OCI images are stored into this persistent volume, by setting the field `spec.storage.enabled` to `true`.
+
+By default, the requested size for the Persistent Volume will be `1Gi`. You can ask to set a specific size with the field `spec.storage.ociRegistryImage`
+
+```bash
+$ cat <<EOF | oc apply -f -
+apiVersion: registry.devfile.io/v1alpha1
+kind: DevfileRegistry
+metadata:
+  name: devfile-registry
+spec:
+  devfileIndex:
+    image: quay.io/devfile/devfile-index:next
+  telemetry:
+    registryName: test
+  storage:
+    enabled: true
+    ociRegistryImage: 2Gi
+EOF
+```
+
+## Configuring TLS for Ingress/Route resource
+
+The operator creates a Route resource (on OpenShift) or an Ingress resources (on Kubernetes)
+to give access to the Web frontend.
+
+By default, the Ingress or Route is secured by TLS.
+
+You can ask the operator to disable the use of TLS by setting the field `spec.tls.enabled` to false.
+
+```bash
+$ cat <<EOF | kubectl apply -f -
+apiVersion: registry.devfile.io/v1alpha1
+kind: DevfileRegistry
+metadata:
+  name: devfile-registry
+spec:
+  devfileIndex:
+    image: quay.io/devfile/devfile-index:next
+  telemetry:
+    registryName: test
+  tls:
+    enabled: false
+EOF
+```
+
+You can ask the operator to configure the TLS with a specific certificate, by specifying a secret
+containing the certificate and the associated private key using the field `spec.tls.ociRegistryImage`:
+
+```bash
+$ kubectl create secret tls my-tls-secret --key=certs/ingress-tls.key --cert=certs/ingress-tls.crt
+
+$ cat <<EOF | kubectl apply -f -
+apiVersion: registry.devfile.io/v1alpha1
+kind: DevfileRegistry
+metadata:
+  name: devfile-registry
+spec:
+  devfileIndex:
+    image: quay.io/devfile/devfile-index:next
+  telemetry:
+    registryName: test
+  tls:
+    enabled: true
+    ociRegistryImage: my-tls-secret
```

Comment:
This should be `secretName` IMO, but the code currently indicates this:

```
	SecretName string `json:"ociRegistryImage,omitempty"`

```

Do we want to change the code directly, or do we want to deprecate it?

Reply:
Preferably we should fix it before documenting it as it's a typo.